MarkDups error processing hard-clipped supplementary alignments

MarkDups v1.1.5 may encounter an error processing hard-clipping supplementary alignments when forming consensus sequences.

This error can impact users where DNA alignment has been performed outside of oncoanalyser if supplementary alignments were hard-clipped rather than soft-clipped.

An example of relevant log messages with stack trace from the error:

```text
01:44:41.884 [Thread-4] [ERROR] error forming consensus: id(null) fragments(2) coords(chr1_4144412_R_chr1_4144584_N) readCounts(PRIMARY=0 complete, MATE=0 pending, MATE_SUPPLEMENTARY=2 ready)
01:44:41.884 [Thread-4] [ERROR] read: id(A00130:175:HGKJNDSX2:1:2271:14118:33646) coords(chr1:4144532-4144633) cigar(74M2D26M50H) mate(chr1:4144262) flags(2145)
01:44:41.885 [Thread-4] [ERROR] read: id(A00130:175:HGKJNDSX2:1:1565:9091:2268) coords(chr1:4144532-4144601) cigar(70M80H) mate(chr1:4144262) flags(2209)
java.lang.ArrayIndexOutOfBoundsException: Index 70 out of bounds for length 70
        at com.hartwig.hmftools.markdups.consensus.ReadParseState.currentBase(ReadParseState.java:53)
        at com.hartwig.hmftools.markdups.consensus.IndelConsensusReads.addElementBases(IndelConsensusReads.java:209)
        at com.hartwig.hmftools.markdups.consensus.IndelConsensusReads.buildIndelComponents(IndelConsensusReads.java:83)
        at com.hartwig.hmftools.markdups.consensus.ConsensusReads.createConsensusRead(ConsensusReads.java:130)
        at com.hartwig.hmftools.markdups.common.DuplicateGroup.popCompletedReads(DuplicateGroup.java:350)
        at com.hartwig.hmftools.markdups.PartitionReader.processDuplicateGroup(PartitionReader.java:321)
        at com.hartwig.hmftools.markdups.PartitionReader.lambda$processIncompleteRead$0(PartitionReader.java:250)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
        at com.hartwig.hmftools.markdups.PartitionReader.processIncompleteRead(PartitionReader.java:250)
        at com.hartwig.hmftools.markdups.PartitionReader.processSamRecord(PartitionReader.java:209)
        at com.hartwig.hmftools.markdups.BamReader.sliceRegion(BamReader.java:58)
        at com.hartwig.hmftools.markdups.PartitionReader.processRegion(PartitionReader.java:123)
        at com.hartwig.hmftools.markdups.PartitionThread.run(PartitionThread.java:61)
```

--- a/README.md
+++ b/README.md
@@ -122,7 +122,7 @@ in the below table.
 
 ## Known issues
 
-- MarkDups may encounter a fatal error processing hard-clipped supplementary alignments ([#37](https://github.com/nf-core/oncoanalyser/issues/37))
+There are currently no known issues.
 
 ## Credits
 

--- a/modules/local/markdups/main.nf
+++ b/modules/local/markdups/main.nf
@@ -3,8 +3,8 @@ process MARKDUPS {
     label 'process_medium'
 
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
-        'https://depot.galaxyproject.org/singularity/hmftools-mark-dups:1.1.5--hdfd78af_1' :
-        'biocontainers/hmftools-mark-dups:1.1.5--hdfd78af_1' }"
+        'https://depot.galaxyproject.org/singularity/hmftools-mark-dups:1.1.7--hdfd78af_0' :
+        'biocontainers/hmftools-mark-dups:1.1.7--hdfd78af_0' }"
 
     input:
     tuple val(meta), path(bams), path(bais)

